src/modm/math/geometry/location_2d.hpp

```diff
@@ -37,39 +36,60 @@ namespace modm
 	class Location2D
 	{
 	public:
-		Location2D();
-
-		Location2D(const Vector<T, 2>& position, const float& orientation);
-
-		Location2D(const T& x, const T& y, const float& orientation);
-
-		inline const Vector<T, 2>&
-		getPosition() const;
-
-		inline const T&
-		getX() const;
-
-		inline const T&
-		getY() const;
-
-		void
-		setPosition(const Vector<T, 2>& point);
-
-		void
-		setPosition(const T& x, const T& y);
+		Vector<T, 2> position;
+		float orientation = 0;
+
+		constexpr Location2D() = default;
+		constexpr Location2D(const Vector<T, 2>& position, const float& orientation)
```

Comment:
Small types like `float` should be passed by value. Then it is passed in a register instead of on the stack on many platforms.